### DISCUSSION TITLE
Hubspot: Contact

### DISCUSSION THREAD

**Discussion by gjhead** [solspace/craft-freeform#1406] [Questions & Answers]
### What happened?

Noticing a possible bug with how the Hubspot "Duplicate Check & Update" is working.  Your [documentation](https://docs.solspace.com/craft/freeform/v5/integrations/hubspot/) notes that it will check if a contact exists, check it against the "email address" as an identifier and if it's a match will update the contact, but if it does not match it will create a new contact.  This does not seem to be working as intended.

I have two different forms, both submitting to Freeform and using the Hubspot integration.  (Note that this is not live currently, but being tested in dev)

On the first form, with a new contact, I can successfully send the info asa new Hubspot contact.

On the second form, when I hit "Submit" on my form, nothing happens.  But in my browser console, I can see an error from hubspot mentioning that the "Contact Already Exists".  At this point, the form is not submitted and no data is sent to Hubspot.

However, I can hit the Submit button a Second time, and then the data *is* submitted.  

I'm not sure how to go about troubleshooting this.  It's either a bug in the integration Duplicate check & Update, or it's possible there may be some additional documentation missing on your side (or that I cannot find?)

### How can we reproduce this?

1.  Submit a form with the Hubspot Integration that pushes to the Contact API
2.  Submit the form a second time with the same contact information.
3. 

### Freeform Edition

Pro

### Freeform Version

5.3.4

### Craft Version

Craft Pro 4.9.7

### When did this issue start?

Unsure

**Comment by kjmartens** [comment]
Hi @gjhead,

I just played around with this now and I can't seem to duplicate the issue you describe. In my testing, it successfully updates the existing user in HubSpot.

Can you tell me more about your setup and flow? Are you just mapping to Contacts only (not deals, companies, etc)? Which fields are you mapping to?

And do you have any sort of automation rules in HubSpot that might be causing this?
https://community.hubspot.com/t5/Lists-Lead-Scoring-Workflows/Manage-duplicate-contacts/m-p/400634

**Reply by gjhead** [reply]
Hey @kjmartens just saw this reply.  Let me take a second look at my setup and get back to you.
